docs/guide/partitioning.rst

```diff
@@ -0,0 +1,214 @@
+
+    duplicate_df = df.copy()
+    duplicate_df.F = "bar"
+
+    dm = store_dataframes_as_dataset(
+        store_factory,
+        "another_partitioned_dataset",
+        [df, duplicate_df],
+        partition_on=["E", "F"],
+    )
+    sorted(dm.partitions.keys())
+
+
+Note that, since 2 dataframes have been provided as input to the function, there are
+4 different files created, even though only 2 different combinations of values of E and
+F are found, ``E=test/F=foo`` and ``E=train/F=foo`` (However, these 4 physical partitions
+can be read as just the 2 logical partitions by using the argument
+``concat_partitions_on_primary_index=True`` at reading time).
+
+For datasets consisting of multiple tables, explicit partitioning on columns can only be
+performed if the column exists in both tables and is of the same data type: guaranteeing
+that their types are the same is part of schema validation in Kartothek.
+
+For example:
+
+.. ipython:: python
+    :okwarning:
+
+    df.dtypes
+    different_df = pd.DataFrame(
+        {"B": pd.to_datetime(["20130102", "20190101"]), "L": [1, 4], "Q": [True, False]}
+    )
+    different_df.dtypes
+
+    dm = store_dataframes_as_dataset(
+        store_factory,
+        "multiple_partitioned_tables",
+        [{"data": {"table1": df, "table2": different_df}}],
+        partition_on="B",
+    )
+
+    sorted(dm.partitions.keys())
+
+
+As noted above, when data is appended to a dataset, Kartothek guarantees it has
+the proper schema and partitioning.
+
+The order of columns provided in ``partition_on`` is important, as the partition
+structure would be different if the columns are in a different order.
+
+.. note:: Every partition must have data for every table. An empty dataframe in this
+          context is also considered as data.
+
+
+Force partitioning by shuffling using Dask
+------------------------------------------
+
+By default, the partitioning logic is applied per physical input partition when writing. In particular, this means that when calling `partition_on` on a column with total N unique values, this may create up to M x N files, where M is the number of physical input partitions.
+
+.. ipython:: python
+    :okwarning:
+
+    import dask.dataframe as dd
+    import numpy as np
+    from kartothek.io.dask.dataframe import update_dataset_from_ddf
+
+    df = pd.DataFrame(
+        {
+            # Good partition column since there are only two unique values
+            "A": [0, 1] * 100,
+            # Too many values for partitioning but still discriminative for querying
+            "B": np.repeat(range(20), 10),
+            "C": "some_payload",
+        }
+    )
+
+    ddf = dd.from_pandas(df, npartitions=10)
+
+    dm = update_dataset_from_ddf(
+        ddf, dataset_uuid="no_shuffle", store=store_factory, partition_on="A", table="table"
+    ).compute()
+    sorted(dm.partitions.keys())
+
+Shuffling
+*********
+
+To circumvent this, we offer a shuffle implementation for dask dataframes.
```

Comment:
> To circumvent this,

what? be specific

Reply:
Rephrased and added more context. Please review again